src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public side1: number,
+    public side2: number,
+    public side3: number,
+  ) {
+    if (this.side1 <= 0 || this.side2 <= 0 || this.side3 <= 0) {
+      throw new Error('Side can not be 0 or less');
+    }
+
+    if (this.side1 >= (this.side2 + this.side3)
+    || this.side2 >= (this.side3 + this.side1)
+    || this.side3 >= (this.side1 + this.side2)) {
+      throw new Error('Sides 1, 2 and 3 can not form a triangle');
+    }
+  }
+
+  getArea(): number {
+    const halfPer = (this.side1 + this.side2 + this.side3) / 2;
```

Comment:
This name is more obvious and meaningful.
```suggestion
    const halfPerimetr = (this.side1 + this.side2 + this.side3) / 2;
```